File: rec-core/src/main/java/net/kimleo/rec/repository/RecConfig.java
package net.kimleo.rec.repository;

import net.kimleo.rec.accessor.Accessor;
import net.kimleo.rec.sepval.parser.ParseConfig;

public interface RecConfig {
    String name();
    ParseConfig parseConfig();
    String key();
    String format();
    Accessor<String> accessor();
    boolean keepOrigin();
}

File: rec-core/src/main/kotlin/net/kimleo/rec/repository/RecConfig.kt
package net.kimleo.rec.repository

import net.kimleo.rec.accessor.Accessor
import net.kimleo.rec.sepval.parser.ParseConfig

interface RecConfig {
    val name: String
    val parseConfig: ParseConfig
    val key: String?
    val format: String
    val accessor: Accessor<String>
}